### ISSUE TITLE
Add ability to use custom ObjectMapper for JSON schema

### ISSUE THREAD

**Issue by onobc** [spring-projects/spring-pulsar#723]
Currently users can not register their own ObjectMapper (or customize the one used by Pulsar) to be used when de/ser message payloads of JSON schema.

Motivation: https://stackoverflow.com/questions/78654478/springboot-pulsar-objectmapper

**Comment by onobc** [comment]
This feature ill be available in Spring Pulsar `1.2.0-M1` but you can get a sneak peek here:
https://github.com/onobc/spring-pulsar/tree/custom-object-mapper

If you want to use this in your application **now** you can do the following:

1. Copy [JsonSchemaUtil](https://github.com/onobc/spring-pulsar/commit/55fb1c9f253c324f03bd6c1ddee395c3b2b13221#diff-df9e8dfa7456388042a2b711d37bc098836d9304cef9a8c85a3a63cd1ba934af) into your project

2. Add the unshaded Jackson lib(s) to your project (so you can provide your own object mapper as that is the whole point of this addition)
    ```groovy
    implementation 'org.springframework.boot:spring-boot-starter-json'
    ```
    or 
    ```groovy
    implementation 'com.fasterxml.jackson.core:jackson-databind'
    ```
3. Register a schema resolver customizer bean that adds a custom schema mapping where the schema is created via JsonSchemaUtil. For example:
      ```java
      @Bean
      SchemaResolver.SchemaResolverCustomizer<DefaultSchemaResolver> schemaResolverCustomizer() {
	      return (DefaultSchemaResolver schemaResolver) -> {
		      var myObjectMapper = ...;
		      var schema = JSONSchemaUtil.schemaForTypeWithObjectMapper(MyMessage.class, myObjectMapper);
		      schemaResolver.addCustomSchemaMapping(MyMessage.class, schema);
	      };
      }
      ```
4. Set your `@PulsarListener` schema type to `schemaType = NONE`. 
> [!NOTE]
> This last step will not be required when the feature is released but if you want to use this in your current project now you will need to adjust this in order for the flow to go through the schema resolver.


With the above in place, when you send/receive `MyMessage` messages the custom schema will be resolved and will in turn use your custom object mapper when de/serializing the messages.

**Comment by onobc** [comment]
Closed via https://github.com/spring-projects/spring-pulsar/commit/70e9e60f39b78771b6b23065269b20dfba6bb243 and https://github.com/spring-projects/spring-pulsar/commit/fe07a7a3fd13e00902c220a91907b6bad505a47b
